From b22c7857a00cbb11791cf70c9c71b9f847b4d24c Mon Sep 17 00:00:00 2001
From: David Reese <david@whatcould.com>
Date: Sat, 7 Feb 2009 14:52:31 -0600
Subject: [PATCH] stop overriding engine view paths

---
 railties/lib/initializer.rb |    5 ++---
 1 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/railties/lib/initializer.rb b/railties/lib/initializer.rb
index e3811dd..15454b2 100644
--- a/railties/lib/initializer.rb
+++ b/railties/lib/initializer.rb
@@ -481,9 +481,8 @@ Run `rake gems:install` to install the missing gems.
     # set to use Configuration#view_path.
     def initialize_framework_views
       if configuration.frameworks.include?(:action_view)
-        view_path = ActionView::Template::Path.new(configuration.view_path)
-        ActionMailer::Base.template_root ||= view_path if configuration.frameworks.include?(:action_mailer)
-        ActionController::Base.view_paths = view_path if configuration.frameworks.include?(:action_controller) && ActionController::Base.view_paths.empty?
+        ActionMailer::Base.view_paths = configuration.view_path if configuration.frameworks.include?(:action_mailer) ActionMailer::Base.view_paths.empty?
+        ActionController::Base.view_paths = configuration.view_path if configuration.frameworks.include?(:action_controller) && ActionController::Base.view_paths.empty?
       end
     end
 
-- 
1.6.0


From c3af86bfb087451b223b4e628932c9713210956a Mon Sep 17 00:00:00 2001
From: David Reese <david@whatcould.com>
Date: Sat, 7 Feb 2009 14:59:25 -0600
Subject: [PATCH] stop overriding engine view paths

---
 railties/lib/initializer.rb |   16 ++++++++--------
 1 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/railties/lib/initializer.rb b/railties/lib/initializer.rb
index 15454b2..ada4bbf 100644
--- a/railties/lib/initializer.rb
+++ b/railties/lib/initializer.rb
@@ -368,13 +368,13 @@ Run `rake gems:install` to install the missing gems.
     end
 
     def load_view_paths
-      if configuration.frameworks.include?(:action_view)
-        if configuration.cache_classes
-          view_path = ActionView::Template::EagerPath.new(configuration.view_path)
-          ActionController::Base.view_paths = view_path if configuration.frameworks.include?(:action_controller)
-          ActionMailer::Base.template_root = view_path if configuration.frameworks.include?(:action_mailer)
-        end
-      end
     end
 
     # Eager load application classes
@@ -481,7 +481,7 @@ Run `rake gems:install` to install the missing gems.
     # set to use Configuration#view_path.
     def initialize_framework_views
       if configuration.frameworks.include?(:action_view)
-        ActionMailer::Base.view_paths = configuration.view_path if configuration.frameworks.include?(:action_mailer) ActionMailer::Base.view_paths.empty?
+        ActionMailer::Base.view_paths ||= configuration.view_path if configuration.frameworks.include?(:action_mailer)
         ActionController::Base.view_paths = configuration.view_path if configuration.frameworks.include?(:action_controller) && ActionController::Base.view_paths.empty?
       end
     end
-- 
1.6.0

